sys/include/senml.h

```diff
@@ -0,0 +1,501 @@
+/*
+ * Copyright (C) 2021 Silke Hofstra
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_senml SenML
+ * @ingroup     sys
+ * @brief       Basic SenML types.
+ *
+ * The `senml` module contains the building blocks for using
+ * [SenML](https://www.rfc-editor.org/rfc/rfc8428).
+ * This module provides the basic types that can be used with (for example)
+ * @ref sys_senml_cbor for encoding measurement data.
+ *
+ * Some attributes defined in SenML need to be enabled explicitly,
+ * see @ref senml_attr_t for details. To enable all attributes, set:
+ *
+ * ```
+ * CFLAGS += -DSENML_ATTR_SUM
+ * CFLAGS += -DSENML_ATTR_VERSION
+ * CFLAGS += -DSENML_ATTR_UPDATE_TIME
```

Comment:
Can you add `ifndef/define` for these and prefix them with `CONFIG_`?

Reply:
I have prefixed them with `CONFIG_`, but I don't understand what you want with an `ifndef/define`?

Reply:
I mean that you never actually defines these, you rely on doing nothing if they are not defined, which works, but does not allow me to search for them in doxygen. So I would like to have something like

```
#ifndef CONFIG_SENML_ATTR_SUM
#define CONFIG_SENML_ATTR_SUM   0
#ifndef
```

(Thar would require switching all the `ifdef  CONFIG_SENML_ATTR_SUM` to `if IS_ACTIVE(CONFIG_SENML_ATTR_SUM)`)

Or otherwise something like:

```
/**
 * @brief   Define SenML update time attribute, disabled by default
 */
#if defined(DOXYGEN)
#define CONFIG_SENML_ATTR_UPDATE_TIME
#endif
```